src/wp-includes/blocks.php

```diff
@@ -850,10 +850,52 @@ function get_hooked_blocks() {
 	return $hooked_blocks;
 }
 
+/**
+ * Adds a list of hooked block types to an anchor block's ignored hooked block types.
+ *
+ * This function is meant for internal use only.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param array                   $parsed_anchor_block The anchor block, in parsed block array format.
+ * @param string                  $relative_position   The relative position of the hooked blocks.
+ *                                                     Can be one of 'before', 'after', 'first_child', or 'last_child'.
+ * @param array                   $hooked_blocks       An array of hooked block types, grouped by anchor block and relative position.
+ * @param WP_Block_Template|array $context             The block template, template part, or pattern that the anchor block belongs to.
+ * @return string An empty string.
+ */
+function set_ignored_hooked_blocks_metadata( &$parsed_anchor_block, $relative_position, $hooked_blocks, $context ) {
+	$anchor_block_type  = $parsed_anchor_block['blockName'];
+	$hooked_block_types = isset( $hooked_blocks[ $anchor_block_type ][ $relative_position ] )
+		? $hooked_blocks[ $anchor_block_type ][ $relative_position ]
+		: array();
+
+	/** This filter is documented in wp-includes/blocks.php */
+	$hooked_block_types = apply_filters( 'hooked_block_types', $hooked_block_types, $relative_position, $anchor_block_type, $context );
+	if ( empty( $hooked_block_types ) ) {
+		return '';
```

Comment:
I don't know what the convention is but do we have to explicitly return a value? E.g. can we not just `return;`?

Reply:
The function that this is part of -- `set_ignored_hooked_blocks_metadata` -- is used as a callback arg for `make_{before|after}_block_visitor`. Its signature needs to be consistent with `insert_hooked_blocks`, which is the default value for the callback.

Previously, when `insert_hooked_blocks` would take care of both inserting hooked blocks _and_ setting the anchor block's `metadata.ignoredHookedBlocks` attribute, it needed a way to do both. I decided to pass the anchor block as a reference, and to have it return the markup generated for the inserted hooked blocks as a string.

It's still doing the latter; and `make_{before|after}_block_visitor` expects a string as a return value, which it'll happily prepend or append to the current block, respectively.

I could maybe change the visitors to allow for `null` or falsey return values, causing the visitor to ignore the return value 🤔 It's just that an empty string was kinda easier to do 😅

Reply:
Ah I see. I think leaving it as is, is fine. My question came from a place of curiosity which this answers. Thank you!